esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1049,25 +1049,81 @@ void WsWuInfo::getInfo(IEspECLWorkunit &info, unsigned long flags)
     getWorkflow(info, flags);
 }
 
+StringBuffer& WsWuInfo::getWorkunitProcessLogPath(const char* process, StringBuffer& path)
+{
+    Owned<IPropertyTreeIterator> procs = cw->getProcesses(process, nullptr);
+    if (!procs->first())
+        return path;
+
+    StringBuffer logSpec;
+    procs->query().getProp("@log", logSpec);
+    if (!logSpec.length())
+        return path;
+
+    splitFilename(logSpec, nullptr, &path, nullptr, nullptr);
+    return path;
+}
+
+unsigned WsWuInfo::getNumberOfSlavesByClusterName()
+{
+    StringAttr clusterName(cw->queryClusterName());
+    if (!clusterName.length()) //Cluster name may not be set yet
+        return 0;
+
+    Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName.str());
+    if (!clusterInfo)
+    {
+        IWARNLOG("Cannot find TargetClusterInfo for workunit %s", cw->queryWuid());
+        return 0;
+    }
+
+    return clusterInfo->getNumberOfSlaveLogs();
+}
+
+unsigned WsWuInfo::getNumberOfSlavesFromZAPFiles()
+{
+    StringBuffer zapWUFolderPath;
+    getWorkunitProcessLogPath("Thor", zapWUFolderPath);
+
+    unsigned numberOfSlaves = 0;
+    Owned<IFile> zapFileFolder = createIFile(zapWUFolderPath);
+    Owned<IDirectoryIterator> thorSlaveFiles = zapFileFolder->directoryFiles("*_thorslave.*");
+    ForEach(*thorSlaveFiles)
+    {
+        const char* pStr = strstr(thorSlaveFiles->query().queryFilename(), "_thorslave.");
+        pStr += 11;
+        const char* slaveNumStrEnd = strchr(pStr, '.');
+        if (!slaveNumStrEnd)
+            continue;
+
+        StringBuffer slaveNumStr;
+        slaveNumStr.append(slaveNumStrEnd - pStr, pStr);
+        unsigned slaveNum = atoi(slaveNumStr);
+        if (slaveNum > numberOfSlaves)
+            numberOfSlaves = slaveNum;
+    }
+    return numberOfSlaves;
+}
+
 unsigned WsWuInfo::getWorkunitThorLogInfo(IArrayOf<IEspECLHelpFile>& helpers, IEspECLWorkunit &info, unsigned long flags, unsigned& helpersCount)
 {
     unsigned countThorLog = 0;
 
     IArrayOf<IConstThorLogInfo> thorLogList;
     if (cw->getWuidVersion() > 0)
     {
-        StringAttr clusterName(cw->queryClusterName());
-        if (!clusterName.length()) //Cluster name may not be set yet
-            return countThorLog;
-
-        Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName.str());
-        if (!clusterInfo)
+        unsigned numberOfSlaveLogs = 0;
+        StringAttr thorMasterLogDateSearchString;
+        if (debugImportedStr.length()) //WU imported from a ZAP report
```

Comment:
As discussed offline, would be good to remove references to 'improted' and ZAP where possible in new code, and instead distinguish where necessary as 'logsPerWorkUnit' flag or similar.
Since we plan to have workunits per workunit in general anyway.

Reply:
Should the coming import() function insert a 'logsPerWorkUnit' flag into the WU tree if not exist?

Reply:
for now I think the 'imported' string you added in the publish info is fine.
But internally refer to that once, and use a flag everywhere for conditional code where necessary.